components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FxaClient.kt

```diff
@@ -0,0 +1,423 @@
+     *
+     * @param scope Single OAuth scope (no spaces) for which the client wants access
+     * @param ttl time in seconds for which the token will be valid
+     * @return [AccessTokenInfo] that stores the token, along with its scopes and keys when complete
+     * @throws FxaException.Unauthorized We couldn't provide an access token
+     * for this scope. The caller should then start the OAuth Flow again with
+     * the desired scope.
+     */
+    suspend fun getAccessToken(
+        scope: String,
+        ttl: Long? = null,
+        requireScopedKey: Boolean = false,
+    ): AccessTokenInfo = wrapMethodCallAndPersist("getAccessToken") {
+        inner.getAccessToken(scope, ttl, requireScopedKey)
+    }
+
+    /**
+     * Tries to return a session token
+     *
+     * FIXME: https://github.com/mozilla/application-services/issues/5819
+     *
+     * @throws FxaException Will send you an exception if there is no session token set
+     */
+    fun getSessionToken(): String {
+        return inner.getSessionToken()
+    }
+
+    /**
+     * Get the current device id
+     *
+     * FIXME: https://github.com/mozilla/application-services/issues/5819
+     *
+     * @throws FxaException Will send you an exception if there is no device id set
+     */
+    fun getCurrentDeviceId(): String {
+        return inner.getCurrentDeviceId()
+    }
+
+    /**
+     * Provisions an OAuth code using the session token from state
+     *
+     * @param authParams Parameters needed for the authorization request
+     * This performs network requests, and should not be used on the main thread.
+     */
+    suspend fun authorizeOAuthCode(
+        authParams: AuthorizationParameters,
+    ): String = wrapMethodCall("authorizeOAuthCode") {
+        inner.authorizeCodeUsingSessionToken(authParams)
+    }
+
+    /**
+     * This method should be called when a request made with
+     * an OAuth token failed with an authentication error.
+     * It clears the internal cache of OAuth access tokens,
+     * so the caller can try to call [getAccessToken] or [getProfile]
+     * again.
+     *
+     * FIXME: https://github.com/mozilla/application-services/issues/5819
+     */
+    fun clearAccessTokenCache() {
+        inner.clearAccessTokenCache()
+    }
+
+    /**
+     * Saves the current account's authentication state as a JSON string, for persistence in
+     * the Android KeyStore/shared preferences. The authentication state can be restored using
+     * [FirefoxAccount.fromJSONString].
+     *
+     * FIXME: https://github.com/mozilla/application-services/issues/5819
+     *
+     * @return String containing the authentication details in JSON format
+     */
+    fun toJSONString(): String {
+        return inner.toJson()
+    }
+
+    /**
+     * Retrieves the list of the connected devices in the current account, including the current one.
+     */
+    suspend fun getDevices(ignoreCache: Boolean = false): Array<Device> = wrapMethodCall("getDevices") {
```

Comment:
mildly surprised we don't want to persist the result in a cache? IOW, I'm surprised this is different to `getProfile()`

Reply:
Good point.  I just translated what the old code did, but I think it should have been persisting the state.